snap/hooks/pre-refresh

```diff
@@ -1,8 +1,20 @@
 #!/bin/sh
 set -eu
 
+INITIAL_PREVIOUS_VERSION_CONFIG="0.2.0"
+
 if [ -n "${DEBUG:-}" ]; then
   logger -t "${SNAP_NAME}" "pre-refresh: Setting previous-version to ${SNAP_VERSION}"
 fi
 
-snapctl set previous-version="${SNAP_VERSION}"
+
+version_less_than() {
+  [ "$(printf '%s\n' "${@}" | sort -V | head -n1)" = "$1" ]
+}
+
+# If we are downgrading, we don't set the `previous-version` if the target version is older
+# than the version that introduced this configuration (0.2.0).
+# Otherwise, we will end up with stale data.
+if version_less_than "${SNAP_VERSION}" "${INITIAL_PREVIOUS_VERSION_CONFIG}"; then
+  snapctl set previous-version="${SNAP_VERSION}"
+fi
```

Comment:
I don't think that's going to work. We only ship this `pre-refresh` hook in 2.0.0, so if it exists, `SNAP_VERSION` will always be >= 2.0.0. I think unsetting the `previous-version` in the `post-refresh` hook, as you suggested yesterday, should work though.

Reply:
Yep you are right. The problem with what I suggested yesterday is that the pre-refresh hook will run when the downgrade happens and it will set the config. Marking the PR as draft for now, until I test things out.

Reply:
>  The problem with what I suggested yesterday is that the pre-refresh hook will run when the downgrade happens and it will set the config.

Ah, right! Too bad.